File: api-tsl/src/tr/gov/tubitak/uekae/esya/api/tsl/model/schemeinformation/TSLType.java
package tr.gov.tubitak.uekae.esya.api.tsl.model.schemeinformation;

import org.w3c.dom.Document;
import org.w3c.dom.Element;

import tr.gov.tubitak.uekae.esya.api.tsl.TSLException;
import tr.gov.tubitak.uekae.esya.api.tsl.model.core.BaseElement;
import tr.gov.tubitak.uekae.esya.api.tsl.util.Constants;
import tr.gov.tubitak.uekae.esya.api.tsl.util.XmlUtil;

public class TSLType extends BaseElement {
	private String mTSLType;

	public TSLType(Document document, String iTSLType) {
		super(document);
		mTSLType = iTSLType;
		mElement.setTextContent(mTSLType);
	}

	public TSLType(Document document) {
		super(document);
	}

	public TSLType(Element aElement) throws TSLException {
		super(aElement);
		mTSLType = XmlUtil.getText(aElement);
	}

	@Override
	public String getLocalName() {
		return Constants.TAG_TSLTYPE;
	}

	@Override
	public String getNamespace() {
		return Constants.NS_TSL;
	}

	public String getTslType() {
		return mTSLType;
	}

	public void setTslType(String value) {
		mTSLType = value;
		mElement.setTextContent(mTSLType);
	}
}

File: api-tsl/csharp/src/model/schemeinformation/TSLType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;
using tr.gov.tubitak.uekae.esya.api.common.util;
using tr.gov.tubitak.uekae.esya.api.tsl.model.core;
using tr.gov.tubitak.uekae.esya.api.tsl.util;

namespace tr.gov.tubitak.uekae.esya.api.tsl.model.schemeinformation
{
    public class TSLType : BaseElement
    {
        private string mTSLType;

        public TSLType(XmlDocument document, string iTSLType) : base(document)
        {
            mTSLType = iTSLType;
            mElement.InnerText = mTSLType;
        }

        public TSLType(XmlDocument document) : base(document)
        {
        }

        public TSLType(XmlElement aElement) : base(aElement)
        {
            mTSLType = XmlUtil.getText(aElement);
        }

        public override string LocalName
        {
            get { return Constants.TAG_TSLTYPE; }
        }

        public override string Namespace
        {
            get
            {
                return Constants.NS_TSL;
            }
        }

        public string TslType
        {
            get { return mTSLType; }
            set 
            { 
                mTSLType = value;
                mElement.InnerText = mTSLType;
            }
        }
    }
}
